### PR TITLE
❇️ Create classes and interfaces for the Product submodule

### PR DESCRIPTION
**Pull request by Doctor-Vee**

### CODE REVIEW

File: paystack-domain/src/main/java/com/chrisworks/paystackclient/domain/product/CreateProductRequest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import com.chrisworks.paystackclient.domain.Amount;
+import com.chrisworks.paystackclient.domain.request.RequestBody;
+
+public class CreateProductRequest implements RequestBody<CreateProductRequest> {
+    private final String name;
+    private final String description;
+    private final String price;
+    private final String currency;
+    private final boolean unlimited;
+    private final Integer quantity;
```
Comment (chriseteka) [inline paystack-domain/src/main/java/com/chrisworks/paystackclient/domain/product/CreateProductRequest.java]: Make the `unlimited` and `quantity` fields optional, also use the builder pattern to declare setters for these fields. Compare what was done in the `UpdatePlanRequest.java` class

File: paystack-domain/src/main/java/com/chrisworks/paystackclient/domain/product/ProductResponse.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    public record Single(boolean status, String message, ProductResponse data)
+            implements PaystackSingleResponse<ProductResponse> {
+    }
+
+    public record Multiple(boolean status, String message, List<ProductResponse> data, PageMetaInfo.Impl meta)
+            implements PaystackMultiResponse<ProductResponse> {
+    }
+
+    private record Metadata(String background_color) {
```
Comment (chriseteka) [inline paystack-domain/src/main/java/com/chrisworks/paystackclient/domain/product/ProductResponse.java]: Remove this class and substitute its usage with `Object` or `Map<String, Object>`

File: paystack-domain/src/main/java/com/chrisworks/paystackclient/domain/product/ProductResponse.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    private record ShippingFields(String delivery_note) {
```
Comment (chriseteka) [inline paystack-domain/src/main/java/com/chrisworks/paystackclient/domain/product/ProductResponse.java]: Same thing goes for the Shipping fields

File: paystack-domain/src/main/java/com/chrisworks/paystackclient/domain/product/UpdateProductRequest.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package com.chrisworks.paystackclient.domain.product;
+
+import com.chrisworks.paystackclient.domain.Amount;
+
+public final class UpdateProductRequest extends CreateProductRequest {
+
```
Comment (chriseteka) [inline paystack-domain/src/main/java/com/chrisworks/paystackclient/domain/product/UpdateProductRequest.java]: Re-use the `CreateProductRequest,` maybe rename it to `CreateOrUpdateProductRequest`

**Review by chriseteka — COMMENTED** [review]
Looks good to me, nice work 😉

File: paystack-domain/src/main/java/com/chrisworks/paystackclient/domain/product/CreateProductRequest.java (hunk 1 above)
Comment (Doctor-Vee) [reply paystack-domain/src/main/java/com/chrisworks/paystackclient/domain/product/CreateProductRequest.java]: Done

File: paystack-domain/src/main/java/com/chrisworks/paystackclient/domain/product/ProductResponse.java (hunk 2 above)
Comment (Doctor-Vee) [reply paystack-domain/src/main/java/com/chrisworks/paystackclient/domain/product/ProductResponse.java]: Done

File: paystack-domain/src/main/java/com/chrisworks/paystackclient/domain/product/ProductResponse.java (hunk 3 above)
Comment (Doctor-Vee) [reply paystack-domain/src/main/java/com/chrisworks/paystackclient/domain/product/ProductResponse.java]: Done

File: paystack-domain/src/main/java/com/chrisworks/paystackclient/domain/product/UpdateProductRequest.java (hunk 4 above)
Comment (Doctor-Vee) [reply paystack-domain/src/main/java/com/chrisworks/paystackclient/domain/product/UpdateProductRequest.java]: Done

**Pull request merged**